docs/containers/docker-compose.md

```diff
@@ -193,6 +193,76 @@ volumes:
+
+> **Note**: If your workspace has `docker-compose.yml` and `docker-compose.override.yml` and no other compose files, then the `docker-compose` command is invoked with no input files and it implicitly uses these files. In this case, no customization is needed.
+### Command Customization
+[Command customization](https://code.visualstudio.com/docs/containers/reference#_command-customization) provides various ways to customize the `compose up` command based on your requirements. The following are few sample command customization for the `compose up` command.
+#### Base file and an override file
+Let's assume your workspace has a base compose file (`docker-compose.yml`) and an override file for each environment (`docker-compose.dev.yml`, `docker-compose.test.yml` and `docker-compose.prod.yml`) and you always compose up with the base file and an override file. In this case the `compose up` command can be customized as in the following example. When the `compose up` command is invoked, the `${configurationFile}` is replaced by the selected file.
+```json
+"docker.commands.composeUp": [
+    {
+        "label": "override",
+        "template": "docker-compose -f docker-compose.yml ${configurationFile}  up -d --build",
+    }
+]
+```
+#### Template matching
+Let's assume you have  different set of input files for each environment. You could define multiple templates with regular expression match, and the selected file name will be matched against this `match` property and the corresponding template will be used.
+```json
+"docker.commands.composeUp": [
+    {
+        "label": "dev-match",
+        "template": "docker-compose -f docker-compose.yml -f docker-compose.debug.yml -f docker-compose.dev.yml up -d --build",
+        "match": "dev"
+    },
+    {
+        "label": "test-match",
+        "template": "docker-compose -f docker-compose.yml -f docker-compose.debug.yml -f docker-compose.test.yml up -d --build",
+        "match": "test"
+    },
+    {
+        "label": "prod-match",
+        "template": "docker-compose -f docker-compose.yml -f docker-compose.release.yml -f docker-compose.prod.yml up -d --build",
+        "match": "prod"
+    }
+]
+```
+
+#### Pick a template at runtime
```

Comment:
The meaning of "runtime" might be a bit unclear here--usually it is reserved for "application run time", not "VS Code run time". Consider something like this instead

### Pick a template when the command is invoked
If you omit the `match` property from command templates, you will be asked which template to use each time `compose up` command is invoked. For example:

Reply:
or "... VS Code will ask you which template to use when you invoke `compose up` command" (to avoid passive voice)

Reply:
Thanks! this is much cleaner.